appcues/src/main/java/com/appcues/statemachine/StateMachine.kt

```diff
@@ -66,8 +66,14 @@ internal class StateMachine(
         appcuesCoroutineScope.launch {
             stateFlow.collect {
                 when (it) {
-                    is BeginningExperience -> config.experienceListener?.experienceStarted(it.experience.id)
-                    is EndingExperience -> config.experienceListener?.experienceFinished(it.experience.id)
+                    is BeginningExperience -> {
+                        startLifecycleTracking()
```

Comment:
wondering if instead of start/stop lifecycle tracking on transitions if we can do it here when we collect state.

Reply:
its a valid concern, but come to think of it now, maybe we can centralize somehow. will keep this in mind for possible next PR